src/librustdoc/html/render/mod.rs

```diff
@@ -4216,11 +4216,8 @@ fn print_sidebar(cx: &Context<'_>, it: &clean::Item, buffer: &mut Buffer, cache:
     let relpath = if it.is_mod() { "../" } else { "" };
     write!(
         buffer,
-        "<script>window.sidebarCurrent = {{\
-                name: \"{name}\", \
-                ty: \"{ty}\", \
-                relpath: \"{path}\"\
-            }};</script>",
+        "<div id=\"sidebar-vars\" data-name=\"{name}\" data-ty=\"{ty}\" data-relpath=\"{path}\">\
```

Comment:
I assume this needs no escaping because identifiers can't contain any characters that are special in HTML attribtues?

Reply:
They were already in the HTML so I assumed it's not required.

Reply:
The escaping required in inline script tags and attributes is different.

Reply:
Makes sense, I'll add the transformation then.

Reply:
I have the feeling there's nothing needed because these parameters are already limited enough, I'm just not certain.

Reply:
After check:

* `relpath` can be either "../" or ""
* `ty` is a type name, so as far as I can tell, there can be no issue with it
* `name` is an item name, so no issue either
* `root_path` is already used in a lot of places in the HTML, so no problem
* `krate` is the crate name, so I guess it's fine too